### PR TITLE
docs: add header pattern documentation

### PR DESCRIPTION
**Pull request by davelinke**
## What?

Adding the header pattern component documentation

## Why?

To guide developers on how to use the header component

## Screenshots/Screen Recordings

header-docs

### CODE REVIEW

File: packages/docs/PropTables/HeaderPropTable.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    required: true,
+  },
+  {
+    name: 'actions',
+    types: 'Array<ActionButtonProps | ActionDropdownProps>',
+    description: 'An array of actions to display in the header. Can include buttons or dropdowns.',
+  },
+  {
+    name: 'backLink',
+    types: 'BackLinkProps',
```
Comment (chanceaclark) [inline packages/docs/PropTables/HeaderPropTable.tsx:19]: I think we need to add `BackLinkProps` to the tabs

File: packages/docs/pages/header.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      <Panel header="Implementation" headerId="implementation">
+        <Fragment key="basic">
+          <Text>
+            To use <Code primary>Header</Code> import the component from the package:
+          </Text>
+          <CodeSnippet>{`import { Header } from '@bigcommerce/big-design-patterns';`}</CodeSnippet>
+        </Fragment>
+        <CodePreview>
+          {/* jsx-to-string:start */}
+          {function Example() {
```
Comment (chanceaclark) [inline packages/docs/pages/header.tsx:47]: We don't really need this function. We can just let it be `<Header ... />`

**Review by golcinho — APPROVED** [review]

File: packages/docs/PropTables/HeaderPropTable.tsx [hunk 3]
```diff
@@ -11,34 +11,59 @@ const headerProps: Prop[] = [
...
+        <NextLink href={{ hash: 'action-button-prop-table', query: { props: 'action-button' } }}>
+          ActionButton
+        </NextLink>
+        {' | '}
+        <NextLink
+          href={{ hash: 'action-dropdown-prop-table', query: { props: 'action-dropdown' } }}
+        >
+          ActionDropdown
+        </NextLink>
+        {'>'}
```
Comment (chanceaclark) [inline packages/docs/PropTables/HeaderPropTable.tsx:26]: We actually haven't done this before, but I figured it was a better approach than our previous iterations.

File: packages/docs/PropTables/HeaderPropTable.tsx [hunk 4]
```diff
@@ -11,34 +11,59 @@ const headerProps: Prop[] = [
...
     description: 'An array of actions to display in the header. Can include buttons or dropdowns.',
   },
   {
     name: 'backLink',
-    types: 'BackLinkProps',
+    types: (
+      <NextLink href={{ hash: 'backlink-prop-table', query: { props: 'back-link' } }}>
+        BackLink
+      </NextLink>
+    ),
```
Comment (chanceaclark) [inline packages/docs/PropTables/HeaderPropTable.tsx:37]: If we are referring to another prop, we need to add links to the prop table.

File: packages/docs/PropTables/HeaderPropTable.tsx [hunk 5]
```diff
@@ -11,34 +11,59 @@ const headerProps: Prop[] = [
...
   {
     name: 'icon',
     types: 'ReactNode',
     description: 'An icon to display to the left of the title.',
   },
-  {
-    name: 'children',
-    types: 'ReactNode',
-    description: 'Any additional content to render inside the header.',
-  },
```
Comment (chanceaclark) [inline packages/docs/PropTables/HeaderPropTable.tsx]: `children` is not actually passed anywhere in the component so I think we'll need to remove the `PropsWithChildren` on the type.

File: packages/docs/pages/header.tsx [hunk 6]
```diff
@@ -85,24 +86,34 @@ const HeaderPage = () => {
...
             },
             {
-              id: 'action-button-props',
+              id: 'action-button',
               title: 'ActionButton',
-              render: () => <ActionButtonPropsTable />,
+              render: () => (
+                <ActionButtonPropsTable
+                  id="action-button-prop-table"
+                  inheritedProps={<ButtonPropTable collapsible />}
```
Comment (chanceaclark) [inline packages/docs/pages/header.tsx:99]: We like to use `inheritedProps` to showcase they are extended. The only issue is that we can't omit things from these tables. How we do props in the documentation needs to be revamped which can be for the future.

File: packages/docs/pages/header.tsx [hunk 7]
```diff
@@ -111,14 +122,14 @@ const HeaderPage = () => {
       <Panel header="Do's and Don'ts" headerId="guidelines">
         <GuidelinesTable
           discouraged={[
-            <>Don’t overload the header with too many actions or content.</>,
-            <>Don’t use more than one primary action button.</>,
-            <>Avoid unrelated or non-contextual content in the header.</>,
+            'Don’t overload the header with too many actions or content.',
```
Comment (chanceaclark) [inline packages/docs/pages/header.tsx:125]: Use strings instead of fragments where possible.

**Review by bc-apostoliuk — APPROVED** [review]

**Pull request merged**